external/graphdb-service.js

```diff
@@ -169,106 +166,123 @@ class GraphdbService {
+                            result.id = quad._subject.id.replace('did:dkg:', '');
+                            break;
+                        case 'http://schema.org/hasTimestamp':
+                            result.metadata.timestamp = JSON.parse(quad._object.id);
+                            break;
+                        case 'http://schema.org/hasUALs':
+                            result.metadata.UALs.push(JSON.parse(quad._object.id));
+                            break;
+                        case 'http://schema.org/hasIssuer':
+                            result.metadata.issuer = JSON.parse(quad._object.id);
+                            break;
+                        case 'http://schema.org/hasVisibility':
+                            result.metadata.visibility = JSON.parse(quad._object.id);
+                            break;
+                        case 'http://schema.org/hasDataHash':
+                            result.metadata.dataHash = JSON.parse(quad._object.id);
+                            break;
+                        case 'http://schema.org/hasKeywords':
+                            result.metadata.keywords.push(JSON.parse(quad._object.id));
+                            break;
+                        case 'http://schema.org/hasSignature':
+                            result.signature = JSON.parse(quad._object.id);
+                            break;
+                        case 'http://schema.org/hasRootHash':
+                            result.rootHash = JSON.parse(quad._object.id);
+                            break;
+                        case 'http://schema.org/hasBlockchain':
+                            result.blockchain.name = JSON.parse(quad._object.id);
+                            break;
+                        case 'http://schema.org/hasTransactionHash':
+                            result.blockchain.transactionHash = JSON.parse(quad._object.id);
+                            break;
+                        default:
+                            break;
                     }
                 } catch (e) {
-                    this.logger.error({ msg: `Error in extracting metadata: ${e}. ${e.stack}`, Event_name: constants.ERROR_TYPE.EXTRACT_METADATA_ERROR });
+                    this.logger.error({
+                        msg: `Error in extracting metadata: ${e}. ${e.stack}`,
+                        Event_name: constants.ERROR_TYPE.EXTRACT_METADATA_ERROR
+                    });
                 }
             }
+
+            result.metadata.keywords.sort();
+            if (!result.metadata.UALs.length)
+                delete result.metadata.UALs;
+            else
+                result.metadata.UALs.sort();
+
+
             accept(result);
         });
     }
 
     async createMetadata(assertion) {
-        const result = await this.toRDF({
+        const metadata = {
             '@context': 'https://www.schema.org/',
+            '@id': `did:dkg:${assertion.id}`,
```

Comment:
Move 'did:dkg' string inside of the constants

Reply:
We are using it only on the graph level, so I guess that it is easier to have it only there